### ISSUE TITLE
Add Register filtering

### ISSUE THREAD

**Issue by allan-on** [opensrp/fhircore#2698]
**Describe the feature request.**
This is a new advanced ability for the user/practitioner to filter the register based on any or all of the supported attributes. 

**Feature mockup**
![image](https://github.com/opensrp/fhircore/assets/16164649/8f35c843-4433-4f84-a1a6-e726d638bf6b)

**Additional context**
See [discussions](https://github.com/opensrp/fhircore/discussions/2138) on this feature 

**Acceptance criteria**

- [x] #2760 
- [x] #2764

**Comment by allan-on** [comment]
Reopening this because only #2760 is done (closed by https://github.com/opensrp/fhircore/pull/2744). 
#2764 is still pending (previously blocked but is now unblocked after the merging of https://github.com/opensrp/fhircore/pull/2718) .

**Comment by allan-on** [comment]
@f-odhiambo @ellykits From the designs shared with the design team, the count of filtered records should be updated in the count badge as seen in the mocks (3). 
@HenryRae @rowo Do we need to update the count showing on the navigation drawer as well? 
In my opinion, having the count on the nav drawer showing the total count of records in that register and the badge showing count of filtered records, should suffice?

**Comment by allan-on** [comment]
The last task 
> Clearing all filters should reload the register and display clients in that register based on the register config dataQueries + filterCriteria

in [Implement filtering of register records #2764 ](https://github.com/opensrp/fhircore/issues/2764) will be completed after the Google FHIR SDK functionality [Clear all action on Questionnaire and functionality to change text on submit action button #2221 ](https://github.com/google/android-fhir/issues/2221) is done

**Comment by rowo** [comment]
@allan-on did you get this question answered? The nav drawer number is the total count for that register. The badge shows the number of filters in place I think but @HenryRae correct me if I'm wrong. On the actual filter screen itself, I think it would be a nice to have to show the matching results once filters are selected (like done on sites like Airbnb), but that's not necessary right now.

**Comment by allan-on** [comment]
Hi, @rowo Thanks for the clarification. 
The current implementation is as you've described.

**Comment by HenryRae** [comment]
[This](https://github.com/opensrp/fhircore/issues/2698#issuecomment-1744084433) is correct, alternatively quick solution would be showing just red dot without count.

**Comment by pld** [comment]
@allan-on what's remaining here now that https://github.com/opensrp/fhircore/issues/2764 is closed?

**Comment by allan-on** [comment]
Hi @pld There is this one remaining bit 
> The last task
> 
> > Clearing all filters should reload the register and display clients in that register based on the register config dataQueries + filterCriteria
> 
> in [Implement filtering of register records #2764 ](https://github.com/opensrp/fhircore/issues/2764) will be completed after the Google FHIR SDK functionality [Clear all action on Questionnaire and functionality to change text on submit action button #2221 ](https://github.com/google/android-fhir/issues/2221) is done

The filters are still usable without this feature, but this issue is still open for tracking that last remaining bit.

**Comment by pld** [comment]
OK yea, cool, will close this that's a separate feature that is useful here and elsewhere
